karate-tests/src/test/java/mocks/sandbox/post-patient.js

```diff
@@ -0,0 +1,114 @@
+  '5954974977', '5157195761', '5897052085', '5359513144', '5446520122',
+  '5103506718', '5468649624', '5002272274', '5941993854', '5574773538',
+  '5899264950', '5604719625', '5117676297', '5705279671', '5890418181'
+]
+
+context.nhsNumberIndex = context.nhsNumberIndex || 0
+
+function generateObjectId () {
+  // generates a random ID for the name and address objects, e.g. 8F1A21BC
+  return Math.random().toString(16).substr(2, 8).toUpperCase()
+}
+
+function getTodaysDate () {
+  // returns today's date in the format YYYY-MM-DD
+  const today = new Date()
+  const year = today.getFullYear()
+  const month = today.getMonth() + 1
+  const day = today.getDate()
+  return `${year}-${month < 10 ? '0' : ''}${month}-${day < 10 ? '0' : ''}${day}`
+}
+
+const NEW_PATIENT = context.read('classpath:mocks/stubs/postPatientResponses/new_patient.json')
+const SINGLE_MATCH = context.read('classpath:mocks/stubs/postPatientResponses/SINGLE_MATCH_FOUND.json')
+const MULTIPLE_MATCHES = context.read('classpath:mocks/stubs/postPatientResponses/MULTIPLE_MATCHES_FOUND.json')
+
+function requestMatchesErrorScenario (request) {
+  // the mocks are programmed to demonstrate error scenarios where we already have matching patients
+  // and we don't create a patient. There is no complex logic here where we try to mimic the matching
+  // logic of the real system, just a simple check that will demonstrate the behaviour when specific
+  // data is sent in the request.
+  let match = false
+  const family = request.body.name['name.familyName']
+  const postalCode = request.body.address['address.postalCode']
+  if (family === 'McMatch-Single' && postalCode === 'BAP4WG') {
+    const body = JSON.parse(JSON.stringify(SINGLE_MATCH))
+    body.issue[0].diagnostics = 'Unable to create new patient. NHS number 5900054586 found for supplied demographic data.'
+    response.body = body
+    match = true
+  } else if (family === 'McMatch-Multiple' && postalCode === 'DN19 7UD') {
+    response.body = MULTIPLE_MATCHES
+    match = true
+  }
+  return match
+}
+
+function postPatientRequestIsValid (request) {
+  // check the request body has the expected structure
+  let valid = true
+  if (!request.body.nhsNumberAllocation) {
+    const body = context.read('classpath:mocks/stubs/errorResponses/MISSING_VALUE.json')
+    body.issue[0].diagnostics = "Missing value - 'nhsNumberAllocation'"
+    response.body = body
+    response.status = 400
+    valid = false
+  }
+  return valid
+}
+
+function userHasPermission (request) {
+  // check the user making the request has permission to create a patient
+  let valid = true
+  if (request.header('Authorization') === 'Bearer APP_RESTRICTED') {
+    const body = context.read('classpath:mocks/stubs/errorResponses/INVALID_METHOD.json')
+    body.issue[0].details.coding[0].display = 'Cannot create resource with application-restricted access token'
+    response.body = body
+    response.status = 403
+    valid = false
+  }
+  return valid
+}
```

Comment:
I think user access requests are also not authorised, does that also need to be here?

Reply:
Just went through the rest of the PR, are you doing these in another ticket?

Reply:
You mean when you're authenticated as a patient? Yes, that will be the subject of another task, for now I'm just covering healthcare worker and app restricted.